service/dynamodb/endpoints.go

```diff
@@ -192,9 +192,7 @@ func finalizeClientEndpointResolverOptions(options *Options) {
 }
 
 func resolveEndpointResolverV2(options *Options) {
-	if options.EndpointResolverV2 == nil {
-		options.EndpointResolverV2 = NewDefaultEndpointResolverV2()
-	}
+	options.EndpointResolverV2 = NewDefaultEndpointResolverV2()
```

Comment:
this looks like were going to overwrite anything EndpointResolverV2 that the user may have set? why?

Reply:
It's can't, this happens before user config fns run.

Reply:
but the user could have set `EndpointResolverV2` directly on the options struct and then call `New`
https://github.com/aws/aws-sdk-go-v2/blob/main/service/dynamodb/api_client.go#L46

Reply:
You're right I always forget about New being used in isolation. Will fix.

Reply:
Fixed in codegen but will actually save regeneration for merge.